### PR TITLE
DDCE-75: Auth Upgrade

### PR DESCRIPTION
**Pull request by Ufuoma-Daniel**

### CODE REVIEW

File: app/controllers/auth/AuthFunctionality.scala [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  lazy val signInUrl: String = ApplicationConfig.ggSignInUrl
+  val origin: String = "ers-returns-frontend"
+
+  def loginParams: Map[String, Seq[String]] = Map(
+    "continue" -> Seq(ExternalUrls.loginCallback),
+    "origin" -> Seq(origin)
+  )
+
+	private def handleException(implicit request: Request[AnyContent]): PartialFunction[Throwable, Result] = {
+		case _: NoActiveSession => Redirect(ApplicationConfig.ggSignInUrl, loginParams)
```
Comment (ericohwotu) [inline app/controllers/auth/AuthFunctionality.scala]: ApplicationConfig.ggSignInUrl defined as signInUrl above

File: project/FrontendBuild.scala [hunk 2]
```diff
@@ -33,6 +33,7 @@ private object AppDependencies {
     "uk.gov.hmrc" %% "domain" % domainVersion,
     "uk.gov.hmrc" %% "http-caching-client" % httpCachingVersion,
     "uk.gov.hmrc" %% "play-language" % "3.4.0",
+		"uk.gov.hmrc" %% "auth-client" % "2.32.0-play-25",
```
Comment (ericohwotu) [inline project/FrontendBuild.scala]: indentation

File: test/controllers/CsvFileUploadControllerSpec.scala [hunk 3]
```diff
@@ -194,28 +179,27 @@ class CsvFileUploadControllerSpec extends UnitSpec with OneAppPerSuite with ERSF
...
+      override val sessionService: SessionService = mock[SessionService]
       override val ersConnector: ErsConnector = mock[ErsConnector]
       override val cacheUtil: CacheUtil = mock[CacheUtil]
 
-      override def showSuccess()(implicit authContext: AuthContext, request: Request[AnyRef], hc: HeaderCarrier): Future[Result] = Future(Ok)
+      override def showSuccess()(implicit authContext: ERSAuthData, request: Request[AnyRef], hc: HeaderCarrier): Future[Result] = Future(Ok)
     }
 
     "redirect for unauthorised users to login page" in {
+			setUnauthorisedMocks()
```
Comment (ericohwotu) [inline test/controllers/CsvFileUploadControllerSpec.scala]: indentation

File: test/controllers/CsvFileUploadControllerSpec.scala [hunk 4]
```diff
@@ -194,28 +179,27 @@ class CsvFileUploadControllerSpec extends UnitSpec with OneAppPerSuite with ERSF
...
       val result = csvFileUploadController.success().apply(FakeRequest("GET", ""))
       status(result) shouldBe SEE_OTHER
       result.header.headers("Location").contains("/gg/sign-in") shouldBe true
     }
 
     "show the result of showUploadFilePage() for authorised users" in {
-      withAuthorisedUser { user =>
+			setAuthMocks()
```
Comment (ericohwotu) [inline test/controllers/CsvFileUploadControllerSpec.scala]: indentation

File: test/controllers/CsvFileUploadControllerSpec.scala [hunk 5]
```diff
@@ -392,18 +381,20 @@ class CsvFileUploadControllerSpec extends UnitSpec with OneAppPerSuite with ERSF
 
   }
 
+	class Setup extends CsvFileUploadController {
+		override val authConnector: PlayAuthConnector = mockAuthConnector
+		override val attachmentsConnector: AttachmentsConnector = mock[AttachmentsConnector]
+		override val sessionService: SessionService = mock[SessionService]
+		override val ersConnector: ErsConnector = mock[ErsConnector]
+		override val cacheUtil: CacheUtil = mock[CacheUtil]
```
Comment (ericohwotu) [inline test/controllers/CsvFileUploadControllerSpec.scala]: indentation

File: test/controllers/CsvFileUploadControllerSpec.scala [hunk 6]
```diff
@@ -413,27 +404,20 @@ class CsvFileUploadControllerSpec extends UnitSpec with OneAppPerSuite with ERSF
...
-
-      override val attachmentsConnector = mock[AttachmentsConnector]
-      override val sessionService = mock[SessionService]
-      override val ersConnector: ErsConnector = mock[ErsConnector]
-      override val cacheUtil: CacheUtil = mock[CacheUtil]
-    }
+    val csvFileUploadController: Setup = new Setup
 
     "redirect for unauthorised users to login page" in {
+			setUnauthorisedMocks()
```
Comment (ericohwotu) [inline test/controllers/CsvFileUploadControllerSpec.scala]: indentation same as auth mocks below

File: test/controllers/FileUploadControllerSpec.scala [hunk 7]
```diff
@@ -60,124 +57,126 @@ class FileUploadControllerSpec extends PlaySpec with OneAppPerSuite
...
+        mockErsConnector.removePresubmissionData(ArgumentMatchers.any[SchemeInfo]())(ArgumentMatchers.any[ERSAuthData](), ArgumentMatchers.any[HeaderCarrier]())
       ) thenReturn Future.successful(HttpResponse(OK))
 
       when(
         mockCacheUtil.fetch[RequestObject](any())(any(), any(), any(), any())
       ) thenReturn Future.successful(ersRequestObject)
 
-      withAuthorisedUser { user =>
-        validationResults(user) { result =>
+			setAuthMocks()
```
Comment (ericohwotu) [inline test/controllers/FileUploadControllerSpec.scala]: indent

File: test/controllers/FileUploadControllerSpec.scala [hunk 8]
```diff
@@ -60,124 +57,126 @@ class FileUploadControllerSpec extends PlaySpec with OneAppPerSuite
...
-      when(mockErsConnector.removePresubmissionData(Matchers.any[SchemeInfo]())(Matchers.any[AuthContext](), Matchers.any[HeaderCarrier]()))
+      when(mockSessionService.retrieveCallbackData()(ArgumentMatchers.any(), ArgumentMatchers.any())).thenReturn(Future.successful(Some(callbackData)))
+      when(mockCacheUtil.fetch[ErsMetaData](ArgumentMatchers.any(), ArgumentMatchers.any())(ArgumentMatchers.any(), ArgumentMatchers.any(), ArgumentMatchers.any())).thenReturn(Future.successful(validErsMetaData))
+      when(mockErsConnector.validateFileData(ArgumentMatchers.any[CallbackData](), ArgumentMatchers.any())(ArgumentMatchers.any[ERSAuthData](), ArgumentMatchers.any(), ArgumentMatchers.any[HeaderCarrier]())).thenReturn(Future.successful(HttpResponse(500)))
+      when(mockErsConnector.removePresubmissionData(ArgumentMatchers.any[SchemeInfo]())(ArgumentMatchers.any[ERSAuthData](), ArgumentMatchers.any[HeaderCarrier]()))
         .thenReturn(Future.successful(HttpResponse(OK)))
-      withAuthorisedUser { user =>
-        validationResults(user) { result =>
+			setAuthMocks()
```
Comment (ericohwotu) [inline test/controllers/FileUploadControllerSpec.scala]: indent

File: test/controllers/FileUploadControllerSpec.scala [hunk 9]
```diff
@@ -60,124 +57,126 @@ class FileUploadControllerSpec extends PlaySpec with OneAppPerSuite
...
-      when(mockErsConnector.removePresubmissionData(Matchers.any[SchemeInfo]())(Matchers.any[AuthContext](), Matchers.any[HeaderCarrier]()))
+      when(mockSessionService.retrieveCallbackData()(ArgumentMatchers.any(), ArgumentMatchers.any())).thenReturn(Future.successful(Some(callbackData)))
+      when(mockCacheUtil.fetch[ErsMetaData](ArgumentMatchers.any(), ArgumentMatchers.any())(ArgumentMatchers.any(), ArgumentMatchers.any(), ArgumentMatchers.any())).thenReturn(Future.successful(validErsMetaData))
+      when(mockErsConnector.validateFileData(ArgumentMatchers.any[CallbackData](), ArgumentMatchers.any())(ArgumentMatchers.any[ERSAuthData](), ArgumentMatchers.any(), ArgumentMatchers.any[HeaderCarrier]())).thenReturn(Future.successful(HttpResponse(OK)))
+      when(mockErsConnector.removePresubmissionData(ArgumentMatchers.any[SchemeInfo]())(ArgumentMatchers.any[ERSAuthData](), ArgumentMatchers.any[HeaderCarrier]()))
         .thenReturn(Future.successful(HttpResponse(INTERNAL_SERVER_ERROR)))
-      withAuthorisedUser { user =>
-        validationResults(user) { result =>
+			setAuthMocks()
```
Comment (ericohwotu) [inline test/controllers/FileUploadControllerSpec.scala]: all setAuthMock indent

File: test/controllers/FileUploadControllerSpec.scala [hunk 10]
```diff
@@ -238,39 +236,38 @@ class FileUploadControllerSpec extends PlaySpec with OneAppPerSuite
...
       }
     }
   }
 
   "calling success" should {
     val fileUploadController: FileUploadController = new FileUploadController {
-      val authConnector = mockAuthConnector
-      override val attachmentsConnector = mock[AttachmentsConnector]
-      override val sessionService = mock[SessionService]
+			val authConnector: PlayAuthConnector = mockAuthConnector
```
Comment (ericohwotu) [inline test/controllers/FileUploadControllerSpec.scala]: indent

File: test/controllers/ReportableEventsControllerTest.scala [hunk 11]
```diff
@@ -167,67 +179,70 @@ class ReportableEventsControllerTest extends UnitSpec with ERSFakeApplicationCon
...
       ).thenReturn(
-        sapRequestRes match {
-          case true => Future.successful("1234567890")
-          case _ => Future.failed(new RuntimeException)
-        }
+        if (sapRequestRes) {
+					Future.successful("1234567890")
+				} else {
+					Future.failed(new RuntimeException)
+				}
```
Comment (ericohwotu) [inline test/controllers/ReportableEventsControllerTest.scala]: indentation

File: test/controllers/CsvFileUploadControllerSpec.scala [hunk 12]
```diff
@@ -167,12 +152,12 @@ class CsvFileUploadControllerSpec extends UnitSpec with OneAppPerSuite with ERSF
...
+        mockCacheUtil.fetch[CsvFilesCallbackList](any(), any())(any(), any(), any())
       ).thenReturn(
         Future.successful(CsvFilesCallbackList(List()))
       )
       when(
-        mockCacheUtil.fetch[RequestObject](anyString())(any(), any(), any(), any())
+        mockCacheUtil.fetch[RequestObject](any())(any(), any(), any(), any())
       ).thenReturn(
         Future.successful(ersRequestObject)
       )
```
[the remark this answers is not in the document]
Comment (vimalsuresh) [reply test/controllers/CsvFileUploadControllerSpec.scala]: will be done as part of refactor ticket by DDCE team.

**Pull request merged**